app/models/manageiq/providers/embedded_ansible/automation_manager/credential.rb

```diff
@@ -26,16 +26,15 @@ def self.raw_create_in_provider(manager, params)
     create!(create_params)
   end
 
-  def self.password_attribute_keys
-    self::API_ATTRIBUTES.map do |k, v|
-      k if v[:type] == :password
-    end.compact
-  end
-
   def self.encrypt_queue_params(params)
-    encrypted_params = params.slice(*password_attribute_keys)
-    encrypted_params.transform_values! { |v| ManageIQ::Password.try_encrypt(v) }
-    params.merge(encrypted_params)
+    options = params.dup
+    DDF.traverse(:fields => self::API_ATTRIBUTES) do |field|
+      key_path = field[:name].try(:split, '.')
```

Comment:
The tests are failing because the keys in the schema are symbols, but the field names are strings. Unfortunately a symbol will never equal a string, even if they have the same value. So the solution for this is to convert the `key_path` to an array of symbols from an array of strings:
```suggestion
      key_path = field[:name].try(:split, '.').try(:map, &:to_sym)
```